### PR TITLE
Fill in Function alias to have a parameter value

### LINKED ISSUE #582: AutoPublishAlias with a !Ref value marked as false positive
*cfn-lint version: 0.11.1*

I'm running into an issue with `AutoPublishAlias`. When using a `!Ref` as its value, I get the following error message:
```
E0001 Resource with id [SkillFunction] is invalid. 'AutoPublishAlias' must be a string or a Ref to a template parameter
foo.yml:1:1
```

Here's the minimal template to reproduce the issue:
```yaml
---

AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31

Parameters:
  Stage:
    Description: Environment stage (deployment phase)
    Type: String
    AllowedValues:
      - beta
      - prod

Resources:
  SkillFunction:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: '.'
      Handler: main.handler
      Runtime: python3.7
      Timeout: 30
      MemorySize: 128
      AutoPublishAlias: !Ref Stage
```

The error message is actually very helpful (👏 for that), but as you can see, I am using a `!Ref` to a template parameter, so this should not cause a linting error.

### PR DESCRIPTION
**Pull request by kddejong**
*Issue #, if available:*
Fix #582
*Description of changes:*
- Pass parameter values to Transform as in some cases they are needed

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/cfnlint/transform.py [hunk 1]
```diff
@@ -70,6 +71,13 @@ def _replace_local_codeuri(self):
             if resource_type == 'AWS::Serverless::Function':
 
                 Transform._update_to_s3_uri('CodeUri', resource_dict)
+                auto_publish_alias = resource_dict.get('AutoPublishAlias')
```
Comment (cmmeyer) [inline src/cfnlint/transform.py]: Is this the only property that allows/requires a fake value for a !Ref? Does it make sense to do this for all properties on a Serverless resource?

File: src/cfnlint/transform.py [hunk 2]
```diff
@@ -70,6 +71,13 @@ def _replace_local_codeuri(self):
...
+                if isinstance(auto_publish_alias, dict):
+                    if len(auto_publish_alias) == 1:
+                        for k, v in auto_publish_alias.items():
+                            if k == 'Ref':
```
Comment (cmmeyer) [inline src/cfnlint/transform.py]: Do we need to handle !Join and !Sub as well?  Might be better to think of this as a general function to populate fake values for all params in a resource (yeah.... I know...)

File: src/cfnlint/transform.py (hunk 1 above)
Comment (kddejong) [reply src/cfnlint/transform.py]: From the SAM docs.
`
AutoPublishAlias             Ref of a CloudFormation Parameter  Alias resources created by SAM uses a LocicalId <FunctionLogicalId+AliasName>. So SAM either needs a string for alias name, or a Ref to template Parameter that SAM can resolve into a string.
`

File: src/cfnlint/transform.py (hunk 2 above)
Comment (kddejong) [reply src/cfnlint/transform.py]: From the SAM Docs.
`
AutoPublishAlias             Ref of a CloudFormation Parameter  Alias resources created by SAM uses a LocicalId <FunctionLogicalId+AliasName>. So SAM either needs a string for alias name, or a Ref to template Parameter that SAM can resolve into a string.
`

File: src/cfnlint/transform.py (hunk 1 above)
Comment (kddejong) [reply src/cfnlint/transform.py]: I don't see any other parameter REFs in https://raw.githubusercontent.com/awslabs/serverless-application-model/master/docs/cloudformation_compatibility.rst

File: src/cfnlint/transform.py (hunk 2 above)
Comment (jlhood) [reply src/cfnlint/transform.py]: @cmmeyer is right. This fix shouldn't be specific to `AutoPublishAlias`. My expectation was the fix should put dummy values in for any required template parameters (no default specified). That will future-proof this against other similar issues.

**Pull request merged**